scalyr_agent/tests/test_profilers.py

```diff
@@ -0,0 +1,185 @@
+        _, data_file_path = tempfile.mkstemp()
+
+        MOCK_CONFIG.enable_profiling = True
+        MOCK_CONFIG.profile_log_name = data_file_path
+
+        self.assertTrue(
+            os.path.isfile(data_file_path), "File %s doesn't exist" % (data_file_path)
+        )
+        self.assertTrue(is_file_path_empty(data_file_path))
+
+        profiler = CPUProfiler(config=MOCK_CONFIG)
+        profiler._data_file_path = data_file_path
+
+        try:
+            profiler._start(MOCK_CONFIG, None)
+        finally:
+            profiler._stop(MOCK_CONFIG, None)
+
+        # Verify data is written on _stop method call
+        self.assertTrue(
+            os.path.isfile(data_file_path), "File %s doesn't exist" % (data_file_path)
+        )
+        self.assertFalse(is_file_path_empty(data_file_path))
+
+
+class MemoryProfilerTestCase(unittest.TestCase):
+    @mock.patch("scalyr_agent.profiler.pympler", None)
+    def test_is_running_pympler_module_not_available(self):
+        MOCK_CONFIG.enable_profiling = True
+
+        profiler = MemoryProfiler(config=MOCK_CONFIG)
+        self.assertFalse(profiler._is_available)
+        self.assertFalse(profiler._is_running())
+
+    @mock.patch("scalyr_agent.profiler.pympler", MOCK_PYMPLER)
+    def test_is_running_yappi_module_available(self):
+        MOCK_CONFIG.enable_profiling = True
+
+        profiler = MemoryProfiler(config=MOCK_CONFIG)
+        self.assertTrue(profiler._is_available)
+        self.assertFalse(profiler._is_running())
+
+    @mock.patch(
+        "scalyr_agent.profiler.PeriodicMemorySummaryCaptureThread",
+        MockPeriodicMemorySummaryCaptureThread,
+    )
+    @mock.patch("scalyr_agent.profiler.pympler", MOCK_PYMPLER)
+    def test_profiling_data_is_written_on_stop(self):
+        # Verify data is written on _stop method call
+        _, data_file_path = tempfile.mkstemp()
+
+        MOCK_CONFIG.enable_profiling = True
+        MOCK_CONFIG.memory_profile_log_name = data_file_path
+
+        self.assertTrue(
+            os.path.isfile(data_file_path), "File %s doesn't exist" % (data_file_path)
+        )
+        self.assertTrue(is_file_path_empty(data_file_path))
+
+        profiler = MemoryProfiler(config=MOCK_CONFIG)
+        self.assertTrue(profiler._is_available)
+        self.assertFalse(profiler._is_running())
+
+        try:
+            profiler._start(MOCK_CONFIG, None)
+            self.assertTrue(profiler._is_running())
+        finally:
+            profiler._stop(MOCK_CONFIG, None)
+
+        self.assertFalse(profiler._is_running())
+
+        # Verify data is written on _stop method call
+        self.assertTrue(
+            os.path.isfile(data_file_path), "File %s doesn't exist" % (data_file_path)
+        )
+        self.assertFalse(is_file_path_empty(data_file_path))
+
+
+def is_file_path_empty(file_path):
+    # type: (str) -> bool
+    """
+    Return true if the provided file path is empty.
+    """
+    with open(file_path, "r") as fp:
```

Comment:
Why not use `os.path.getsize`?

Reply:
In the future, I also wanted to assert on the content, but in the mean time using ``os.path.getsize`` should work as well.

Will push that change.

Reply:
6377cd2